### ISSUE TITLE
LUA not working after 1.5.18 update

### ISSUE THREAD

**Issue by artmeans-sd** [FrSkyRC/ETHOS-Feedback-Community#4698]
X18se. All working well on ethos 1.5.12

Updated ethos to current (.18) and noticed that one of my LUA scripts wasn’t available on the screen, even though it was still in the scripts directory

I see a new section in the manual (image attached) about selecting LUA on a per model basis – which is a new tab on the model page. But this tab isnt available on my radio - 
 
![image](https://github.com/user-attachments/assets/b843a662-2ec7-402c-b5ed-c3a2ae9e8e85)

**Comment by filduch** [comment]
I would say that the script is the problem. There's regularly changes in the LUA specifications and the script has to be modified accordingly (for example now return "nil" and before "0") I don't know why you don't see the LUA icon.

**Comment by bsongis-frsky** [comment]
You don't attach the script, how do you want I can help you?

**Comment by artmeans-sd** [comment]
![image](https://github.com/user-attachments/assets/6e163c14-53f7-471b-a86d-cdaea752e2ab)
I believe the issue is that the radio is missing the LUA tab in the model page? - As shown in this image
Script works fine in 1.5.12
[F5B caller.zip](https://github.com/user-attachments/files/17620882/F5B.caller.zip)

**Comment by bsongis** [comment]
The small asterisk in the top bar should give you some information

**Comment by artmeans-sd** [comment]
yes, understood on the asterisk - a work in progress.    Bigger issue is why the LUA tab is not available on my radio, in the model menu?

**Comment by artmeans-sd** [comment]
![image](https://github.com/user-attachments/assets/efa99d28-445c-4f7c-9ce1-70c059c14188)

**Comment by Dsrenger** [comment]
The Lua tab is just there, if you have a Lua that triggers the Tab to appear.

There are different LUA script types. For example LUAs that creates widgets. or LUA scripts like the stabilisation LUA for the SR receivers. These widgets do not create the LUA menu tab.

The LUA menu tab will only be created if you have a source or task LUA script.
I havent looked at your script, but I bet its not a task or source script.

So, everything fine with your radio.

The manual is not very clear on this. I messaged Lother some time ago about this. I guess he has it on his radar :)

The reason why your script isnt working anymore should be found in the script itself. There were some changes/bug fixes in the LUA script area.

**Comment by filduch** [comment]
There's no buttom because therer's no valid scripting running I believe. If you touch on the red indicator, you will see where's the error and what's wrong

**Comment by strgaltdel** [comment]
as described in the ethos-manual-screenshot you showed and dennis mentioned:
you attached a widget-lua,
so the "lua-tab" wont be triggered

in case the widget is not running, the author must adopt it to .18 requirements

nil values: 
#4650

handler problems
#4670
